### ISSUE TITLE
false positive in variable-name rule

### ISSUE THREAD

**Issue by HamletDRC** [palantir/tslint#599]
There is a false positive in the variable-name issue. 

My code: 
```ts
    var localStorageItem = ...;
    try {
        JSON.parse(localStorageItem);
     } catch (SyntaxError) {
         return null;
     }
```

SyntaxError gives the error: 
 variable name must be in camelcase or uppercase

It should not.

**Comment by adidahiya** [comment]
you're assigning the error value to a new var named `SyntaxError`... what do you expect instead? do you want to disable this rule for all `catch` statements?

#522 is probably the right way to deal with this

**Comment by HamletDRC** [comment]
You are correct, this is not a false positive. 

I was reading SyntaxError as a class reference, not as a variable reference. I blame too much time coding Java. Sorry for the false issue.
